app/models/document.rb

```diff
@@ -256,6 +256,14 @@ def withdraw
     end
   end
 
+  def discard
+    handle_remote_error { discard! }
+  end
+
+  def discard!
+    Services.publishing_api.discard_draft(content_id, previous_version: previous_version)
+  end
```

Comment:
I don't think we need raise variants of those methods yet, so I'll remove `discard!`. We might want them eventually, but simpler is better for now. 👍 

I think I'll rename `publish!` to `publish` too.